include/assimp/matrix3x3.inl

```diff
@@ -101,16 +101,34 @@ inline aiMatrix3x3t<TReal> aiMatrix3x3t<TReal>::operator* (const aiMatrix3x3t<TR
 
 // ------------------------------------------------------------------------------------------------
 template <typename TReal>
-inline TReal* aiMatrix3x3t<TReal>::operator[] (unsigned int p_iIndex)
-{
-    return &this->a1 + p_iIndex * 3;
+inline TReal* aiMatrix3x3t<TReal>::operator[] (unsigned int p_iIndex) {
+    switch ( p_iIndex ) {
+        case 0:
+            return &a1;
```

Comment:
This is not a sufficient fix. Since a1-a3 are not in the same array, dereferencing the returned pointer can only access a1 safely. Anything else is UB. Likewise for the other 2 cases. It would probably be best to remove this method entirely.